// @mui/x-charts/internals/plugins/featurePlugins/useChartInteraction/useChartInteraction.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useChartInteraction = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _useEventCallback = _interopRequireDefault(require("@mui/utils/useEventCallback"));
var _fastObjectShallowCompare = require("@mui/x-internals/fastObjectShallowCompare");
const useChartInteraction = ({
  store
}) => {
  const cleanInteraction = (0, _useEventCallback.default)(function cleanInteraction() {
    store.set('interaction', (0, _extends2.default)({}, store.state.interaction, {
      pointer: null,
      item: null
    }));
  });
  const removeItemInteraction = (0, _useEventCallback.default)(function removeItemInteraction(itemToRemove) {
    const prevItem = store.state.interaction.item;
    if (!itemToRemove) {
      // Remove without taking care of the current item
      if (prevItem !== null) {
        store.set('interaction', (0, _extends2.default)({}, store.state.interaction, {
          item: null
        }));
      }
      return;
    }
    if (prevItem === null || Object.keys(itemToRemove).some(key => itemToRemove[key] !== prevItem[key])) {
      // The current item is already different from the one to remove. No need to clean it.
      return;
    }
    store.set('interaction', (0, _extends2.default)({}, store.state.interaction, {
      item: null
    }));
  });
  const setItemInteraction = (0, _useEventCallback.default)(function setItemInteraction(newItem, context) {
    if (!(0, _fastObjectShallowCompare.fastObjectShallowCompare)(store.state.interaction.item, newItem)) {
      store.set('interaction', (0, _extends2.default)({}, store.state.interaction, {
        lastUpdate: context.interaction,
        item: newItem
      }));
    }
  });
  const setPointerCoordinate = (0, _useEventCallback.default)(function setPointerCoordinate(coordinate) {
    store.set('interaction', (0, _extends2.default)({}, store.state.interaction, {
      pointer: coordinate,
      lastUpdate: coordinate !== null ? 'pointer' : store.state.interaction.lastUpdate
    }));
  });
  return {
    instance: {
      cleanInteraction,
      setItemInteraction,
      removeItemInteraction,
      setPointerCoordinate
    }
  };
};
exports.useChartInteraction = useChartInteraction;
useChartInteraction.getInitialState = () => ({
  interaction: {
    item: null,
    pointer: null,
    lastUpdate: 'pointer'
  }
});
useChartInteraction.params = {};
