// @mui/x-charts/BarChart/BarLabel/BarLabelPlot.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BarLabelPlot = BarLabelPlot;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));
var _BarLabelItem = require("./BarLabelItem");
var _barClasses = require("../barClasses");
var _jsxRuntime = require("react/jsx-runtime");
const _excluded = ["processedSeries", "skipAnimation"];
/**
 * @ignore - internal component.
 */
function BarLabelPlot(props) {
  const {
      processedSeries,
      skipAnimation
    } = props,
    other = (0, _objectWithoutPropertiesLoose2.default)(props, _excluded);
  const {
    seriesId,
    data
  } = processedSeries;
  const classes = (0, _barClasses.useUtilityClasses)();
  const barLabel = processedSeries.barLabel ?? props.barLabel;
  if (!barLabel) {
    return null;
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("g", {
    className: classes.seriesLabels,
    "data-series": seriesId,
    children: data.map(({
      xOrigin,
      yOrigin,
      x,
      y,
      dataIndex,
      color,
      value,
      width,
      height,
      layout
    }) => /*#__PURE__*/(0, _jsxRuntime.jsx)(_BarLabelItem.BarLabelItem, (0, _extends2.default)({
      seriesId: seriesId,
      dataIndex: dataIndex,
      value: value,
      color: color,
      xOrigin: xOrigin,
      yOrigin: yOrigin,
      x: x,
      y: y,
      width: width,
      height: height,
      skipAnimation: skipAnimation ?? false,
      layout: layout ?? 'vertical'
    }, other, {
      barLabel: barLabel,
      barLabelPlacement: processedSeries.barLabelPlacement || 'center'
    }), dataIndex))
  }, seriesId);
}
