// @mui/x-charts/esm/internals/plugins/featurePlugins/useChartInteraction/useChartInteraction.js
import _extends from "@babel/runtime/helpers/esm/extends";
import useEventCallback from '@mui/utils/useEventCallback';
import { fastObjectShallowCompare } from '@mui/x-internals/fastObjectShallowCompare';
export const useChartInteraction = ({
  store
}) => {
  const cleanInteraction = useEventCallback(function cleanInteraction() {
    store.set('interaction', _extends({}, store.state.interaction, {
      pointer: null,
      item: null
    }));
  });
  const removeItemInteraction = useEventCallback(function removeItemInteraction(itemToRemove) {
    const prevItem = store.state.interaction.item;
    if (!itemToRemove) {
      // Remove without taking care of the current item
      if (prevItem !== null) {
        store.set('interaction', _extends({}, store.state.interaction, {
          item: null
        }));
      }
      return;
    }
    if (prevItem === null || Object.keys(itemToRemove).some(key => itemToRemove[key] !== prevItem[key])) {
      // The current item is already different from the one to remove. No need to clean it.
      return;
    }
    store.set('interaction', _extends({}, store.state.interaction, {
      item: null
    }));
  });
  const setItemInteraction = useEventCallback(function setItemInteraction(newItem, context) {
    if (!fastObjectShallowCompare(store.state.interaction.item, newItem)) {
      store.set('interaction', _extends({}, store.state.interaction, {
        lastUpdate: context.interaction,
        item: newItem
      }));
    }
  });
  const setPointerCoordinate = useEventCallback(function setPointerCoordinate(coordinate) {
    store.set('interaction', _extends({}, store.state.interaction, {
      pointer: coordinate,
      lastUpdate: coordinate !== null ? 'pointer' : store.state.interaction.lastUpdate
    }));
  });
  return {
    instance: {
      cleanInteraction,
      setItemInteraction,
      removeItemInteraction,
      setPointerCoordinate
    }
  };
};
useChartInteraction.getInitialState = () => ({
  interaction: {
    item: null,
    pointer: null,
    lastUpdate: 'pointer'
  }
});
useChartInteraction.params = {};
